src/runtime_src/core/common/api/bo_int.h

```diff
@@ -47,11 +57,13 @@ create_bo(const xrt::hw_context& hwctx, size_t sz, use_type type);
 //
 // Configure the buffer object to be used for debug, dtrace, log, debug queue
 // purpose based on buffer type. The buffer object is tied to a slot using
-// the hw ctx that is used to create the bo. A map of uc or column index and
+// the hw ctx passed to this call. If the hwctx passed in null then hwctx that is
+// used to create the bo is used. A map of uc or column index and
```

Comment:
- What happens if config() is called with ctx_handle, but unconfig() is not?   
- What happens if config() is called but unconfig() is not?
- What happens if unconfig() is called, but config() was not?

It is a very error prone design.  It would be much better if config_bo() returned an RAII object that managed what needs to be manged so that calling unconfig() is not required.   In fact if any of the bullet questions result in unrecoverable error, then we must make changes to fix this.

Reply:
Hi Soren, create_bo returns xrt::bo object. config_bo is just a bo_int function that can be called with xrt::bo argument.
config_bo doesn't return any object.
And also in xrt::bo shim handle destruction we are checking if its a special bo (log, debug, dtrace) and calling unconfig_bo.
We have provided unconfig_bo internal api so that we have more flexibility to unconfigure the address at any point instead of relying on bo destructor call.